packages/nimble-components/src/rich-text/editor/tests/rich-text-editor-mention.spec.ts

```diff
@@ -0,0 +1,824 @@
+            );
+
+            expect(pageObject.getEditorTagNames()).toEqual(['P', 'A', 'A']);
+            expect(pageObject.getEditorLeafContents()).toEqual([
+                'user:1',
+                'test:2'
+            ]);
+            mappingElements[0]!.key = 'test:2';
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getMarkdownRenderedTagNames()).toEqual([
+                'P',
+                RICH_TEXT_MENTION_USERS_VIEW_TAG,
+                RICH_TEXT_MENTION_USERS_VIEW_TAG,
+                'BR'
+            ]);
+            expect(
+                pageObject.getEditorMentionViewAttributeValues('mention-label')
+            ).toEqual(['username1', 'testname2']);
+        });
+
+        it('updating valid `key` in mapping mention should update it to absolute link if it is a mention before', async () => {
+            element.setMarkdown('<user:1> <test:2>');
+            const { mappingElements } = await appendUserMentionConfiguration(
+                element,
+                [{ key: 'user:1', displayName: 'username1' }]
+            );
+            await appendTestMentionConfiguration(element, [
+                { key: 'test:2', displayName: 'testname2' }
+            ]);
+
+            expect(pageObject.getMarkdownRenderedTagNames()).toEqual([
+                'P',
+                RICH_TEXT_MENTION_USERS_VIEW_TAG,
+                RICH_TEXT_MENTION_USERS_VIEW_TAG,
+                'BR'
+            ]);
+            expect(
+                pageObject.getEditorMentionViewAttributeValues('mention-label')
+            ).toEqual(['username1', 'testname2']);
+            mappingElements[0]!.key = 'invalid';
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getEditorTagNames()).toEqual(['P', 'A', 'A']);
+            expect(pageObject.getEditorLeafContents()).toEqual([
+                'user:1',
+                'test:2'
+            ]);
+        });
+    });
+
+    it('should fire "mention-update" event from configuration elment when there is @mention in editor', async () => {
+        const { userMentionElement } = await appendUserMentionConfiguration(
+            element
+        );
+        const mentionUpdateSpy = jasmine.createSpy('mention-update');
+        userMentionElement.addEventListener('mention-update', mentionUpdateSpy);
+        await pageObject.setEditorTextContent('@test');
+        expect(mentionUpdateSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should fire "mention-update" event from configuration elment when there is update in @mention in editor', async () => {
+        const { userMentionElement } = await appendUserMentionConfiguration(
+            element
+        );
+        const mentionUpdateSpy = jasmine.createSpy('mention-update');
+        userMentionElement.addEventListener('mention-update', mentionUpdateSpy);
+        await pageObject.setEditorTextContent('@test');
+        expect(mentionUpdateSpy).toHaveBeenCalledTimes(1);
+    });
```

Comment:
I may be missing something, but these tests look identical other than their names. For the update test ideally you'd set the editor content to `@test`, then create the spy and add the event listener, then append/type text immediately after @test, and assert that the event fired